backend/src/main/java/gov/cdc/usds/simplereport/db/repository/DeviceSpecimenRepository.java

```diff
@@ -0,0 +1,31 @@
+package gov.cdc.usds.simplereport.db.repository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.Query;
+
+import gov.cdc.usds.simplereport.db.model.DeviceSpecimen;
+import gov.cdc.usds.simplereport.db.model.DeviceType;
+import gov.cdc.usds.simplereport.db.model.SpecimenType;
+
+public interface DeviceSpecimenRepository extends EternalEntityRepository<DeviceSpecimen> {
+
+    @Override
+    @EntityGraph(attributePaths = { "deviceType", "specimenType" })
+    @Query(BASE_QUERY + " and e.deviceType.isDeleted = false and e.specimenType.isDeleted = false")
+    public List<DeviceSpecimen> findAll();
+
+    @EntityGraph(attributePaths = { "deviceType", "specimenType" })
+    @Query(BASE_QUERY + " and e.deviceType = :deviceType and e.specimenType = :specimenType")
```

Comment:
No, if the caller is handing in an object as a query parameter they're responsible for deciding if they care about its soft-deletion state.

Reply:
what's the value of users being able to find deleted `DeviceSpecimen`s? i worry it only gives them the ability to do invalid things (like accidentally assign a deleted `DeviceSpecimen` to a `TestEvent` after a refactor), and yields no obvious benefits.

Reply:
This seems like a different question from the original one—this method doesn't return deleted DeviceSpecimenTypes, but it may return a DeviceSpecimenType related to a deleted DeviceType or SpecimenType. 

To answer your question, though, finding a deleted DeviceSpecimenType is useful if you need to undelete it, but probably not otherwise.